packages/core/src/actions/execute.ts

```diff
@@ -1137,6 +1137,62 @@ export const sortSigners = (arr: Array<TreeSigner>): void => {
   })
 }
 
+/**
+ * This is a utility function that is helpful for recovering from a deployment that has timed out. It fetches
+ * all the transaction receipts related to a deployment using the expected events and returns them. If this
+ * function is used when some of the transaction receipts are already tracked, they can be passed in and this
+ * function will return an array of all of the unique transactions.
+ *
+ * This function returns standard ethers.TransactionReceipt objects instead of our SphinxTransactionReceipt
+ * type because that is more flexible and generally useful.
+ */
+export const fetchExecutionTransactionReceipts = async (
+  receipts: ethers.TransactionReceipt[],
+  moduleAddress: string,
+  merkleRoot: string,
+  provider: SphinxJsonRpcProvider | HardhatEthersProvider
+) => {
+  const module = new ethers.Contract(moduleAddress, SphinxModuleABI, provider)
+
+  const SphinxMerkleRootApprovedFilter =
+    module.filters.SphinxMerkleRootApproved(merkleRoot)
+  const SphinxMerkleRootCanceledFilter =
+    module.filters.SphinxMerkleRootCanceled(merkleRoot)
+  const SphinxMerkleRootFailedFilter =
+    module.filters.SphinxMerkleRootFailed(merkleRoot)
+  const SphinxActionSucceededFilter =
+    module.filters.SphinxActionSucceeded(merkleRoot)
+  const SphinxActionFailedFilter = module.filters.SphinxActionFailed(merkleRoot)
+
+  const filters = [
+    SphinxMerkleRootApprovedFilter,
+    SphinxMerkleRootCanceledFilter,
+    SphinxMerkleRootFailedFilter,
+    SphinxActionSucceededFilter,
+    SphinxActionFailedFilter,
+  ]
+
+  const txHashes = receipts.map((r) => r.hash)
+
+  for (const filter of filters) {
+    const latestBlock = await provider.getBlockNumber()
+    const startingBlock = latestBlock - 1999 > 0 ? latestBlock - 1999 : 0
+    const events = await module.queryFilter(filter, startingBlock, latestBlock)
```

Comment:
Not sure if this is a concern or out of scope of this PR, but chains like Arbitrum can produce 2000 blocks in a matter of minutes. E.g. [this block](https://arbiscan.io/block/184139961) on Arbitrum occurs 2000 blocks before [this block](https://arbiscan.io/block/184141961), and they're 9 mins apart. Could this potentially prevent us from retrieving the receipts earlier in the deployment here?

Reply:
[I created a ticket for it.](https://linear.app/chugsplash/issue/CHU-630/use-higher-block-query-block-limit-when-fetching-deployment)